compiler/rustc_middle/src/ty/vtable.rs

```diff
@@ -83,7 +83,17 @@ pub(super) fn vtable_allocation_provider<'tcx>(
         let idx: u64 = u64::try_from(idx).unwrap();
         let scalar = match entry {
             VtblEntry::MetadataDropInPlace => {
-                let instance = ty::Instance::resolve_drop_in_place(tcx, ty);
+                let ty_erased = if let (true, TyKind::FnPtr(poly_fn_sig)) =
+                    (ty.has_late_bound_regions(), ty.kind())
+                {
+                    let re_erased = tcx.erase_late_bound_regions(*poly_fn_sig);
+                    let rebound = Binder::dummy(re_erased);
+                    let reassembled = tcx.mk_fn_ptr(rebound);
+                    reassembled
+                } else {
+                    ty
+                };
+                let instance = ty::Instance::resolve_drop_in_place(tcx, ty_erased);
```

Comment:
More simply, should `resolve_drop_in_place` call `tcx.erase_regions` on its argument?

Reply:
This is something I thought about adding to the list of _open questions_, but forgot as soon as I wrote the first one 🐟.